addons/adyen_platforms/models/adyen_account.py

```diff
@@ -213,75 +284,91 @@ def action_create_redirect(self):
-            'name': _('Transactions'),
-            'view_mode': 'tree,form',
-            'domain': [('adyen_account_id', '=', self.id)],
-            'res_model': 'adyen.transaction',
-            'type': 'ir.actions.act_window',
-            'context': {'group_by': ['adyen_payout_id']}
-        }
+        action = self.env['ir.actions.actions']._for_xml_id('adyen_platforms.adyen_transaction_action')
+        action['domain'] = expression.AND([[('adyen_account_id', '=', self.id)], literal_eval(action.get('domain', '[]'))])
+        return action
 
     def _upload_photo_id(self, document_type, content, filename):
-        self._adyen_rpc('upload_document', {
+        test_mode = self.env['ir.config_parameter'].sudo().get_param('adyen_platforms.test_mode')
+        self._adyen_rpc('v1/upload_document', {
             'documentDetail': {
                 'accountHolderCode': self.account_holder_code,
                 'documentType': document_type,
                 'filename': filename,
+                'description': 'PASSED' if test_mode else '',
             },
             'documentContent': content.decode(),
         })
 
-    def _format_data(self):
+    def _format_data(self, values):
+        fields = set(values.keys())
         data = {
-            'accountHolderCode': self.account_holder_code,
-            'accountHolderDetails': {
-                'address': {
-                    'country': self.country_id.code,
-                    'stateOrProvince': self.state_id.code or None,
-                    'city': self.city,
-                    'postalCode': self.zip,
-                    'street': self.street,
-                    'houseNumberOrName': self.house_number_or_name,
-                },
-                'email': self.email,
-                'fullPhoneNumber': self.phone_number,
-            },
-            'legalEntity': 'Business' if self.is_business else 'Individual',
+            'accountHolderCode': values.get('account_holder_code') or self.account_holder_code,
         }
 
-        if self.is_business:
-            data['accountHolderDetails']['businessDetails'] = {
-                'legalBusinessName': self.legal_business_name,
-                'doingBusinessAs': self.doing_business_as,
-                'registrationNumber': self.registration_number,
-            }
-        else:
-            data['accountHolderDetails']['individualDetails'] = {
-                'name': {
-                    'firstName': self.first_name,
-                    'lastName': self.last_name,
-                    'gender': 'UNKNOWN',
-                },
-                'personalData': {
-                    'dateOfBirth': str(self.date_of_birth),
-                }
+        # *ALL* the address fields are required if one of them changes
+        address_fields = {'country_id', 'state_id', 'city', 'zip', 'street', 'house_number_or_name'}
+        if address_fields & fields:
+            data.setdefault('accountHolderDetails', {})
```

Comment:
Just create a `details` dict at the toplevel, fill it, and then add it to the `data` dict if it's non-empty at the end. Having `data.setdefault('accountHolderDetail')` everywhere is unnecessary.